content/ADR-289-explorer-client-scenes-signed-fetch.md

```diff
@@ -0,0 +1,137 @@
+---
+layout: adr
+adr: 289
+title: Explorer Client Scenes Signed Fetch
+date: 2025-11-07
+status: Living
+type: Standards Track
+spdx-license: CC0-1.0
+authors:
+  - lpetaccio
+  - pentreathm
+  - nachomazzara
+  - gonpombo8
+---
+
+## Abstract
+
+This document describes how the Signed Fetch functionality works in the Explorer Client for Decentraland scenes. It details the structure and purpose of the scene-specific metadata headers that are included when scenes perform HTTP requests. The Explorer Client extends the Signed Fetch mechanism defined in [ADR-44](/adr/ADR-44) with additional contextual metadata including scene identity, parcel location, realm information and a specific signer, the `decentraland-kernel-scene` enabling external services to verify the origin and context of requests.
+
+## Context, Reach & Prioritization
+
+### Context
+
+The Explorer Client runs scenes that interact with external HTTP services. When scenes make requests on behalf of users, additional context beyond the basic Signed Fetch mechanism is valuable for external services. The Explorer Client includes scene-specific metadata in signed requests to provide:
+
+- Scene identification and location (parcel coordinates)
+- Realm information (where the user is connected)
+- Environment context (production, staging, development)
+- Request body integrity verification
+- Signer identification (to identify who made the request)
+
+This additional metadata enables external services to verify not only the user's identity but also the scene context from which the request originates.
+
+**Important**: The Explorer Client controls the entire signature generation process, ensuring that all metadata accurately reflects the true context of the request and cannot be forged by malicious scene code.
```

Comment:
This is accurate; however, I believe it is ensured by the SDK sandbox itself rather than the client, right?

Reply:
It's being ensured by the client's code, as it's the client who builds the request, but I'm not sure if that is something that can be taken into consideration as the SDK sandbox. @gonpombo8 what do you think about that?